sdks/java/extensions/protobuf/src/main/java/org/apache/beam/sdk/extensions/protobuf/ProtoSchemaTranslator.java

```diff
@@ -268,13 +287,12 @@ private static FieldType beamFieldTypeFromSingularProtoField(
           default:
             fieldType = FieldType.row(getSchema(protoFieldDescriptor.getMessageType()));
         }
+        // all messages are nullable in Proto
```

Comment:
Are they? I didn't think that was true. What happens if you declare a submessage required in proto2?

Reply:
You're right. I've corrected the Translator for messages and also added tests.